slu/slu/dev/prompt_setup.py

```diff
@@ -0,0 +1,269 @@
+        for lang in config.get_supported_languages():
+            if NLS_LABEL not in nls_labels[const.NLS_LANG_MAPPING[lang]]:
+                logger.debug(f"nls-key  {NLS_LABEL} not found for lang {lang}")
+            elif not nls_labels[const.NLS_LANG_MAPPING[lang]][NLS_LABEL]:
+                logger.debug(f"Prompt not found for lang {lang}, nls-key {NLS_LABEL}")
+            else:
+                if (
+                    isinstance(nls_labels[const.NLS_LANG_MAPPING[lang]][NLS_LABEL], str)
+                    and len(nls_labels[const.NLS_LANG_MAPPING[lang]][NLS_LABEL]) > 0
+                ):
+                    nls_df.at[i, lang] = nls_labels[const.NLS_LANG_MAPPING[lang]][
+                        NLS_LABEL
+                    ]
+                if (
+                    isinstance(
+                        nls_labels[const.NLS_LANG_MAPPING[lang]][NLS_LABEL], list
+                    )
+                    and len(nls_labels[const.NLS_LANG_MAPPING[lang]][NLS_LABEL]) == 1
+                ):
+                    nls_df.at[i, lang] = nls_labels[const.NLS_LANG_MAPPING[lang]][
+                        NLS_LABEL
+                    ][0]
+
+    return nls_df
+
+
+def validate(df: pd.DataFrame) -> pd.DataFrame:
+
+    if not const.NLS_LABEL in df.columns:
+        raise Exception(f"Mandatory column missing {const.NLS_LABEL}")
+    if not const.STATE in df.columns:
+        raise Exception(f"Mandatory column missing {const.STATE}")
+
+    df = df[(df[const.NLS_LABEL].notna()) & (df[const.STATE].notna())]
+    invalid_rows = df[(df[const.NLS_LABEL].isna()) | (df[const.STATE].isna())]
+    logger.debug(f"Num invalid rows: {invalid_rows.shape[0]}")
+
+
+def get_prompts_map(df: pd.DataFrame) -> pd.DataFrame:
+
+    prompts_map: dict = dict()
+    missing_prompts_map: dict = dict()
+    supported_languages: list = [
+        col
+        for col in df.columns
+        if (_valid_string(col) and col not in [const.NLS_LABEL, const.STATE])
+    ]
+    nls_labels: set = set()
+
+    if not supported_languages:
+        raise Exception("No languages found in the dataset")
+
+    logger.debug(f"Found languages: {supported_languages}")
+    for lang in supported_languages:
+        prompts_map[lang] = {}
+        missing_prompts_map[lang] = []
+
+        for i in tqdm(range(df.shape[0]), desc=f"Fetching prompts for {lang}"):
+            nls_label = df.iloc[i][const.NLS_LABEL]
+            nls_labels.add(nls_label)
+            prompt = df.iloc[i][lang]
+
+            if _valid_string(prompt):
+                prompt = preprocess_prompt(
+                    prompt, fill_token=const.PROMPT_NOISE_FILLER_TOKEN
+                )
+                if _valid_string(prompt):
+                    prompts_map[lang][nls_label] = prompt
+
+    for lang in supported_languages:
+        missing_prompts_map[lang].append(
+            list(nls_labels - set(prompts_map[lang].keys()))
+        )
+
+    return prompts_map, missing_prompts_map
```

Comment:
returns do not match the return type in function signature.

Reply:
Added a new signature

Reply:
That's the wrong way of writing function signature. The correct way would be `Tuple[Dict[<define type here>], Dict[<define type here>]`. 
Check this out - https://stackoverflow.com/questions/37087457/difference-between-defining-typing-dict-and-dict